File: projects/quarkus-course/getting-started/src/main/java/tech/donau/course/GreetingResource.java
package tech.donau.course;

import java.util.UUID;

import javax.inject.Inject;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.HeaderParam;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.MediaType;

import tech.donau.course.service.GreetingService;

@Path("/hello")
public class GreetingResource {
	
	@Inject
	private GreetingService greetingService;
	
	
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello(@QueryParam("name") String name) {
        return greetingService.sayHello(name);
    }
    
}

File: projects/quarkus-course/getting-started-kotlin/src/main/kotlin/tech/donau/course/GreetingResource.kt
package tech.donau.course

import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.inject.Inject
import javax.enterprise.inject.Default
import javax.ws.rs.PathParam

@Path("/hello")
class GreetingResource {
	
	@Inject
	lateinit var greetingService: GreetingService
	
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "hello"
	
	@GET
	@Path("/greeting/{name}")
	@Produces(MediaType.APPLICATION_JSON)
	fun greet(@PathParam("name") name: String): Greeting {
		return Greeting(greetingService.greet(name));
	}
}